### PR TITLE
[Feature] [Doc] Add oracle jdbc catalog

### LINKED ISSUE #35691: Need JDBC Catalog to increase support for Oracle
- [x] https://github.com/StarRocks/starrocks/pull/47517
- [x] https://github.com/StarRocks/starrocks/pull/47465
- [x] https://github.com/StarRocks/starrocks/pull/44887
Please consider adding support for Oracle in the JDBC Catalog. It is urgent to do so

### PR DESCRIPTION
**Pull request by wmtbnbo**
## Why I'm doing:
  [Need JDBC Catalog to increase support for Oracle #35691](https://github.com/StarRocks/starrocks/issues/35691)
  [请求JDBC CATALOG中增加oracle和sqlserver数据源类型 #30704](https://github.com/StarRocks/starrocks/issues/30704)
  [JDBC Catalogs for ORacle #30061](https://github.com/StarRocks/starrocks/issues/30061)

## What I'm doing:
Add oracle jdbc catalog

Fixes https://github.com/StarRocks/starrocks/issues/35691 https://github.com/StarRocks/starrocks/issues/30704
https://github.com/StarRocks/starrocks/issues/30061

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr needs user documentation (for new or modified features or behaviors)
- [x] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wmtbnbo** [comment]
> …om_lib directory avoid overwriting custom dependencies during upgrade
> 
> ## Why I'm doing:
> [Need JDBC Catalog to increase support for Oracle #35691](https://github.com/StarRocks/starrocks/issues/35691) [请求JDBC CATALOG中增加oracle和sqlserver数据源类型 #30704](https://github.com/StarRocks/starrocks/issues/30704) [JDBC Catalogs for ORacle #30061](https://github.com/StarRocks/starrocks/issues/30061)
> 
> ## What I'm doing:
> Add oracle jdbc catalog and Optimize custom dependency loading
> 
> Fixes #issue
> 
> ## What type of PR is this:
> * [ ]  BugFix
> * [x]  Feature
> * [ ]  Enhancement
> * [ ]  Refactor
> * [ ]  UT
> * [x]  Doc
> * [ ]  Tool
> 
> Does this PR entail a change in behavior?
> 
> * [] Yes, this PR will result in a change in behavior.
> * [x]  No, this PR will not result in a change in behavior.
> 
> If yes, please specify the type of change:
> 
> * [ ]  Interface/UI changes: syntax, type conversion, expression evaluation, display information
> * [ ]  Parameter changes: default values, similar parameters but with different default values
> * [ ]  Policy changes: use new policy to replace old one, functionality automatically enabled
> * [ ]  Feature removed
> * [ ]  Miscellaneous: upgrade & downgrade compatibility, etc.
> 
> ## Checklist:
> * [ ]  I have added test cases for my bug fix or my new feature
> * [ ]  This pr needs user documentation (for new or modified features or behaviors)
> * [ ]  I have added documentation for my new feature or new function
> * [ ]  This is a backport pr
> 
> ## Bugfix cherry-pick branch check:
> * [ ]  I have checked the version labels which the pr will be auto-backported to the target branch
>   
>   * [x]  3.3
>   * [x]  3.2
>   * [x]  3.1
>   * [ ]  3.0
>   * [ ]  2.5

[Feature] Add oracle jdbc catalog and default jdbc catalog #43663

File: bin/start_backend.sh [hunk 1]
```diff
@@ -141,6 +141,10 @@ export LIBHDFS_OPTS=$final_java_opt
 # Otherwise, JVM will overwrite the signal handlers for SIGINT and SIGTERM.
 export LIBHDFS_OPTS="$LIBHDFS_OPTS -Xrs"
 
+# add custom_libs to CLASSPATH
+for f in $STARROCKS_HOME/custom_lib/*.jar; do
+    CLASSPATH=$f:${CLASSPATH}
+done
```
Comment (zombee0) [inline bin/start_backend.sh:147]: why we need this?

File: bin/start_fe.sh [hunk 2]
```diff
@@ -163,6 +163,10 @@ fi
 for f in $STARROCKS_HOME/lib/*.jar; do
   CLASSPATH=$f:${CLASSPATH};
 done
+# add custom_libs to CLASSPATH
+for f in $STARROCKS_HOME/custom_lib/*.jar; do
+    CLASSPATH=$f:${CLASSPATH}
+done
```
Comment (zombee0) [inline bin/start_fe.sh:169]: ditto

File: bin/start_backend.sh (hunk 1 above)
Comment (wmtbnbo) [reply bin/start_backend.sh:147]: When using a higher version of Oracle dependency, a character set exception will be reported due to the lack of orai18n. jar dependency as follows:
"Non supported character set (add orai18n.jar in your classpath): ZHS16GBK"
In order to make upgrading operations more user-friendly for users, we have chosen to add this directory.

Comment (wmtbnbo) [reply bin/start_backend.sh:147]: > why we need this?

When using a higher version of Oracle dependency, a character set exception will be reported due to the lack of orai18n. jar dependency as follows:
"Non supported character set (add orai18n.jar in your classpath): ZHS16GBK"
In order to make upgrading operations more user-friendly for users, we have chosen to add this directory.

File: bin/start_fe.sh (hunk 2 above)
Comment (wmtbnbo) [reply bin/start_fe.sh]: > ditto

start_fe.sh and start_backend.sh load custom_lib commit removed

**Review by zombee0 — APPROVED** [review]

**Review by WholeWorld-Timothy — APPROVED** [review]

**Review by DorianZheng — DISMISSED** [review]

**Review by Youngwb — DISMISSED** [review]

**Pull request merged**